src/main.js

```diff
@@ -1,84 +1,73 @@
             'tabClass': '.Concertina-tab',
             'tabActiveClass': '.is-selected',
             'panelClass': '.Concertina-panel',
             'panelActiveClass': '.is-visible'
         };
-        this.options = extendConfig.bind(this)(this.defaults, options);
-        this.tabs = this.element.querySelectorAll(this.options.tabClass);
-        this.panels = this.element.querySelectorAll(this.options.panelClass);
+
+        for (let prop in customOptions) {
+            this.options[prop] = customOptions[prop];
+        }
+
+        this.tabs = [...this.element.querySelectorAll(this.options.tabClass)];
+        this.panels = [...this.element.querySelectorAll(this.options.panelClass)];
     }
 
     init() {
         //TODO: Set aria-controls attr on tabs and aria-labelledby attr on panels.
+
         this.setTabsInactive();
         this.setPanelsInactive();
 
         this.element.addEventListener('click', (e) => {
-            if (e.target.classList.contains(this.options.tabClass.replace('.', ''))){
+            if (this.tabs.includes(e.target)){
                 e.preventDefault();
-
-                this.setTabsInactive();
-                this.setPanelsInactive();
                 this.setTabState(e.target);
-                this.setPanelState(e.target.getAttribute('href').replace('#', ''));
+                this.setPanelState(e.target.getAttribute('href'));
             }
         });
     }
 
-    // Disable all tabs.
-    setTabsInactive() {
-        for (let i = 0; i < this.tabs.length; i += 1) {
-            this.tabs[i].setAttribute('aria-selected', 'false');
-            this.tabs[i].classList.remove(this.options.tabActiveClass.replace('.', ''));
-        }
-    }
-
     // Set state on a single tab.
     setTabState(el) {
         this.setTabsInactive();
+        this.toggleState(el, true);
+    }
+
+    // Set state on a single panel.
+    setPanelState(id) {
+        id = id.replace('#', '');
+        let el = this.panels.find(elem => elem.id == id);
+        this.setPanelsInactive();
+        this.toggleState(el, true);
+    }
 
-        el.setAttribute('aria-selected', 'true');
-        el.classList.add(this.options.tabActiveClass.replace('.', ''));
+    // Disable all tabs.
+    setTabsInactive() {
+        this.tabs.map(elem => this.toggleState(elem, false));
     }
 
     // Disable all panels.
     setPanelsInactive() {
-        for (let i = 0; i < this.panels.length; i += 1) {
-            this.panels[i].setAttribute('aria-hidden', 'true');
-            this.panels[i].classList.remove(this.options.panelActiveClass.replace('.', ''));
-        }
+        this.panels.map(elem => this.toggleState(elem, false));
     }
 
-    // Set state on a single panel.
-    setPanelState(id) {
-        let el = document.getElementById(id);
-
-        el.setAttribute('aria-hidden', 'false');
-        el.classList.add(this.options.panelActiveClass.replace('.', ''));
+    // Toggle the active state.
+    toggleState(el, state) {
```

Comment:
The generic state toggler function allows any method to toggle state without having to repeat the lengthy class update chain. (If we're not using Jquery then we should at least try to keep the vanilla code as readable as possible)